vm/src/obj/objrange.rs

```diff
@@ -319,78 +319,12 @@ impl PyRange {
     fn getitem(&self, subscript: RangeIndex, vm: &VirtualMachine) -> PyResult {
         match subscript {
             RangeIndex::Slice(slice) => {
-                let range_start = self.start.as_bigint();
-                let range_step = self.step.as_bigint();
-                let _tmp_len = self.length();
-                let range_length = _tmp_len.as_bigint();
-
-                let substep = if let Some(slice_step) = slice.step_index(vm)? {
-                    if slice_step.is_zero() {
-                        return Err(vm.new_value_error("slice step cannot be zero".to_string()));
-                    }
-                    slice_step
-                } else {
-                    BigInt::one()
-                };
-
-                let negative_step = substep.is_negative();
-                let lower_bound = if negative_step {
-                    -BigInt::one()
-                } else {
-                    BigInt::zero()
-                };
-                let upper_bound = if negative_step {
-                    &lower_bound + range_length
-                } else {
-                    range_length.clone()
-                };
-
-                let substart = if let Some(slice_start) = slice.start_index(vm)? {
-                    if slice_start.is_negative() {
-                        let tmp = slice_start + range_length;
-                        if tmp < lower_bound {
-                            lower_bound.clone()
-                        } else {
-                            tmp.clone()
-                        }
-                    } else if slice_start > upper_bound {
-                        upper_bound.clone()
-                    } else {
-                        slice_start.clone()
-                    }
-                } else if negative_step {
-                    upper_bound.clone()
-                } else {
-                    lower_bound.clone()
-                };
-
-                let substop = if let Some(slice_stop) = slice.stop_index(vm)? {
-                    if slice_stop.is_negative() {
-                        let tmp = slice_stop + range_length;
-                        if tmp < lower_bound {
-                            lower_bound.clone()
-                        } else {
-                            tmp.clone()
-                        }
-                    } else if slice_stop > upper_bound {
-                        upper_bound.clone()
-                    } else {
-                        slice_stop.clone()
-                    }
-                } else if negative_step {
-                    lower_bound.clone()
-                } else {
-                    upper_bound.clone()
-                };
-
```

Comment:
The 3 lines below do need to remain here, though.

Reply:
The function is equivalent to this CPython code (but the variables are named opposite) https://github.com/python/cpython/blob/master/Objects/rangeobject.c#L306-L322